interpreter/cling/include/cling/Interpreter/Exception.h

```diff
@@ -19,9 +21,10 @@ namespace clang {
 namespace cling {
   ///\brief Base class for all interpreter exceptions.
   ///
-  class InterpreterException {
+  class InterpreterException : public std::exception {
   public:
     virtual const char* what() const throw();
+    virtual void diagnose() const throw();
```

Comment:
Can we make this inline? Please remove the `throw()` at least from this function :-)